service/core/kafka/server/src/main/scala/com/lightbend/lagom/internal/broker/kafka/Producer.scala

```diff
@@ -161,29 +174,25 @@ private[lagom] object Producer {
       Flow[Message].map { message =>
         ProducerMessage.Message(new ProducerRecord[String, Message](topicId, keyOf(message), message), NotUsed)
       } via {
-        ReactiveProducer.flow(producerSettings(serviceLocatorUri))
+        ReactiveProducer.flow(producerSettings(endpoints))
       }
     }
 
-    private def producerSettings(serviceLocatorUri: Option[URI]): ProducerSettings[String, Message] = {
+    private def producerSettings(endpoints: String): ProducerSettings[String, Message] = {
       val keySerializer = new StringSerializer
 
-      val baseSettings = ProducerSettings(context.system, keySerializer, serializer)
-        .withProperty("client.id", self.path.toStringWithoutAddress)
+      val baseSettings =
+        ProducerSettings(context.system, keySerializer, serializer)
+          .withProperty("client.id", self.path.toStringWithoutAddress)
 
-      serviceLocatorUri match {
-        case Some(uri) =>
-          baseSettings.withBootstrapServers(s"${uri.getHost}:${uri.getPort}")
-        case None =>
-          baseSettings.withBootstrapServers(kafkaConfig.brokers)
-      }
+      baseSettings.withBootstrapServers(endpoints)
```

Comment:
Strange, I can't comment on your previous comment. I will add it here.

Yes, I had a commit with a `ServiceLookup` ADT. Later I realised that I could remove it by also resolving it much earlier in the stack. 

The only drawback is that later, when logging failures, we have to do some branching using the service name.